src/charm.py

```diff
@@ -79,16 +70,13 @@ def _mongodb_relation_joined(self, event) -> None:
             try:
                 self._initialise_replica_set()
             except (ConnectionFailure, ConfigurationError):
-                self.unit.status = BlockedStatus(
-                    "failed to initialise replica set")
+                self.unit.status = BlockedStatus("failed to initialise replica set")
                 return
 
         # verify replica set is ready
         if not self._mongo.is_ready():
-            logger.debug(
-                "Replica set for units: %s, is not ready", self._unit_ips)
-            self.unit.status = BlockedStatus(
-                "failed to initialise replica set")
+            logger.debug("Replica set for units: %s, is not ready", self._unit_ips)
+            self.unit.status = BlockedStatus("failed to initialise replica set")
```

Comment:
Should this be Blocked, or Waiting? Feels like the case that warrants a blocked status is handled above -- blocked should only be used in the "HALP I NEED A HUMAN" case :)

Reply:
Yes! You are correct, Balbir and I discussed this in a meeting along with other changes that might want to be made to replication. We were thinking that they would be done in a the next PRs. Is it okay if I fix this in those PRs?

Reply:
Sounds good!